### ISSUE TITLE
Indentation issue with (yas-indent-line 'fixed) and field at start of line

### ISSUE THREAD

**Issue by AndrewKWinn** [joaotavora/yasnippet#906]
## Summary
Under these conditions:
* `yas-indent-line` set to `fixed`
* emacs set to `python-mode`
* snippet keyword is indented
* snippet has an un-indented field (i.e. located at the first column) in the snippet definitions

The field will properly indent all of the text, but *will place the cursor for the un-indented field at the first column, **not** at column of the indented snippet keyword*.

### Bash Commands to setup minimum working example of issue:
```bash
$ emacs --version
GNU Emacs 25.3.1
$ cd /tmp/
$ git clone https://github.com/joaotavora/yasnippet.git yasnippet-bug
$ cd yasnippet-bug/
$ git log -1 --oneline
3d7cafc Make yas-snippet-mode-buffer-p robust against symlinks
$ HOME=$PWD emacs -L .
```
### YASnippet file:
```bash
# -*- mode: snippet -*-
# name: returns
# key: returns
# expand-env: ((yas-indent-line 'fixed))
# --
Returns
-------
${1:type}
    ${2:description}
```
### Emacs pre-expansion (with `python-mode` and `yas-global-mode`)
![image](https://user-images.githubusercontent.com/36708984/36498556-9e7b9520-170c-11e8-9597-8b38918530fb.png)

### Emacs post expansion
![image](https://user-images.githubusercontent.com/36708984/36498081-4bcdccea-170b-11e8-8813-761b4353d3da.png)

Note the cursor position of the above image.  The cursor should be at the beginning of `type`, not at the start of the line.

**Comment by npostavs** [comment]
Ah, I had fixed this sort of thing for the `auto` indent mode, but nobody noticed the same problem for `fixed` until now. Please try the patch in #908.

**Comment by AndrewKWinn** [comment]
While the cursor is now correctly placed,  the inserted spaces before `description` are no longer maintained:

![image](https://user-images.githubusercontent.com/36708984/36981780-04e005a0-205c-11e8-9bf8-994877ad235b.png)

**Comment by npostavs** [comment]
Oops, I must have some whitespace blindness. Should be fixed in 15761e85d63f5ce52a4f163dba93a0561a67b778 now.

**Comment by AndrewKWinn** [comment]
Looks great, thanks!
